### PR TITLE
Work with ContractSpecs in Other Languages

### LINKED ISSUE #606: Create guides for working with contract specs in other programming languages besides JavaScript
**Get awarded:** up to $400 worth of XLM
**Section:** How-To Guides - Conventions: https://github.com/stellar/stellar-docs/tree/main/docs/smart-contracts/guides/conventions

**Guide could include:**

- Introduction that includes what a contract spec typically includes, explaining how these details guide the interaction with contracts through whatever programming language you’re using
- Prerequisites such as the tools, software, or libraries needed
- Tutorials on interacting with Soroban smart contracts in whatever programming language you’re using with code examples, including reading from a contract, writing to a contract, and handling events emitted from a smart contract
- Advanced topics such as upgrading contracts, interacting with multiple contracts, and more

### PR DESCRIPTION
**Pull request by Myestery**
meant to fix #606

### CODE REVIEW

**Comment by briwylde08** [comment]
Thank you so much! I'll get this assigned next week. Our reviews are a little behind so I think you in advance for your patience!

File: docs/build/guides/conventions/working-with-contract-specs.mdx [hunk 1]
```diff
@@ -0,0 +1,1269 @@
...
+
+### 1. Invoking a Contract Function with no Parameters
+
+We will be using the `increment` function of the sample [increment contract](https://github.com/stellar/soroban-examples/tree/main/increment) contract to exemplify this. The `increment` function takes no parameters and increments the counter by 1.
+
+In this scenario, there is no need for conversions and passing the value `null` as contract arguments is sufficient in most SDKs.
+
+<CodeExample>
+
+```rust
```
[the remark this answers is not in the document]
Comment (Myestery) [reply docs/build/guides/conventions/working-with-contract-specs.mdx:249]: The examples show how to achieve the same function in the three SDKs being used throughout the guide 
Screenshot 2024-08-06 at 19 10 56

File: docs/build/guides/conventions/working-with-contract-specs.mdx [hunk 2]
```diff
@@ -0,0 +1,1264 @@
+---
+title: Work with contract specs in Java, Python, and PHP
+description: A guide to understanding and interacting with Soroban smart contracts in different programming languages.
+hide_table_of_contents: false
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/working-with-contract-specs.mdx:4]: This line can just be removed entirely. it defaults to `false`, so there's no need to specify that.

File: docs/build/guides/conventions/working-with-contract-specs.mdx [hunk 3]
```diff
@@ -0,0 +1,1264 @@
...
+
+## Prerequisites
+
+Before diving into contract interactions, ensure you have the following:
+
+- Stellar CLI ([`stellar`](../../smart-contracts/getting-started/setup.mdx#install-the-stellar-cli)) installed
+- A Soroban-compatible SDK for your programming language. View the [list of available SDKs](../../../tools/sdks/library.mdx) to find one that suits your needs
+- Access to a Stellar [RPC server](../../../learn/fundamentals/networks.mdx) (local or on a test network)
+
+For this guide, we will focus on the Java, Python, and PHP SDKs for reference, but the concepts can also be applied to other languages.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/working-with-contract-specs.mdx:29]: Would be really nice to have `Java`, `Python`, and `PHP` as links to the respective SDK documentation.

File: docs/build/guides/conventions/working-with-contract-specs.mdx [hunk 4]
```diff
@@ -0,0 +1,1264 @@
...
+- `map`: Key-value map
+- `symbol`: A small string used mainly for function names and map keys
+
+In this guide and the SDKs, these types are represented as `ScU32`, `ScU64`, `ScI32`, `ScI64`, `ScU128`, `ScI128`, `ScBool`, `ScString`, `ScVec`, `ScAddress`, `ScMap`, and `ScSymbol` respectively.
+
+Every other complex type can be derived using these basic types but these types do not really map to values in the programming languages. The Stellar SDKs provide helper classes to work with these types.
+
+## Working with native Soroban types
+
+One of the most common tasks when working with Soroban smart contracts is converting between Soroban types and native types in your programming language. In this guide, we will go over some common conversions and show how they can be used to invoke contracts with the help of the contract spec.
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/working-with-contract-specs.mdx:231]:
```suggestion
One of the most common tasks when working with Stellar smart contracts is converting between Stellar smart contract types and native types in your programming language. In this guide, we will go over some common conversions and show how they can be used to invoke contracts with the help of the contract spec.
```

File: docs/build/guides/conventions/working-with-contract-specs.mdx [hunk 5]
```diff
@@ -0,0 +1,1264 @@
...
+                        ]
+                    ),
+                    scval.to_symbol("timestamp"): scval.to_uint64(12346),
+                }
+            ),
+        ],
+    )
+    .build()
+)
+# tx.sign(kp)
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/working-with-contract-specs.mdx:1142]: Is this meant to be here?

**Review by ElliotFriend — COMMENTED** [review]
Hi, there @Myestery! This is a **very good** submission! Thanks for the work you've put into it, and for presenting it in a way that makes sense, and walks the user through a lot of really well described, well selected examples.

I've left a couple notes throughout (most quite minor), and took the liberty of changing some link formatting and code indentation here and there.

This is really nice guide, and I think it'll be a really valuable addition to our docs! Thank you!!

File: docs/build/guides/conventions/working-with-contract-specs.mdx (hunk 5 above)
Comment (Myestery) [reply docs/build/guides/conventions/working-with-contract-specs.mdx:1142]: I used it to test my implementation when writing the guide.
Taking it away now

**Comment by Myestery** [comment]
Thanks @ElliotFriend for the review
I have made the corrections as requested

**Review by ElliotFriend — APPROVED** [review]
Looks great! Thanks again!!

**Comment by briwylde08** [comment]
Thanks @Myestery! I'll merge this one and let's take the award conversation back to the corresponding issue. Yay!

**Pull request merged**